contracts/base/AaveV2Wrapper.sol

```diff
@@ -0,0 +1,201 @@
+
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    //----------------------------//
+    //          Events            //
+    //----------------------------//
+
+    event AddAsset(address indexed _asset, address _oracle);
+    event BorrowRatioChanged(uint256 indexed ratio);
+    event WithdrawEvent(
+        address indexed asset,
+        address indexed account,
+        uint256 amount
+    );
+    event DepositEvent(address indexed account, uint256 amount);
+    event CreateToken(
+        address indexed tokenAddress,
+        string tokenName,
+        string tokenSymbol,
+        uint8 decimal
+    );
+
+    //----------------------------//
+    //       View Functions       //
+    //----------------------------//
+
+    /**
+     * @dev fetch the latest price of the asset from oracle
+     * @param oracle address for which we need price
+     */
+    function getLatestPrice(address oracle)
+        internal
+        view
+        returns (int256 price)
+    {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(oracle);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function normalize(uint256 _amount) internal pure returns (uint256) {
+        return _amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    function calculateBorrow(address reserve, uint256 amount)
+        internal
+        view
+        returns (uint256)
+    {
+        if (reserve == ETH) reserve = WETH;
+        int256 assetPrice = getLatestPrice(oracles[reserve]);
+        uint256 availableBorrowAmountIn18Decimals = (normalize(uint256(assetPrice).mul(amount).div(1e18)));
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals.div(1e12);
+        uint256 aaveLTV = availableBorrowAmountIn6Decimals.mul(aaveDefaulLTV).div(BASE);
+        return aaveLTV.mul(borrowRatio).div(BASE);
+    }
+
+    //-----------------------------//
+    // Internal Mutation Functions //
+    //-----------------------------//
+
+    function depositETH(uint256 amountToAave,uint256 amountToCruize) internal {
+        WETHGATEWAY.depositETH{value: amountToAave}(address(POOL), address(this), 0);
+        payable(cruize_wallet).transfer(amountToCruize);
+    }
+
+    function depositERC20(address reserve, uint256 amountToAave,uint256 amountToCruize) internal {
+        IERC20 token = IERC20(reserve);
+        require(token.transferFrom(msg.sender, address(this), amountToAave));
+        require(token.approve(address(POOL), amountToAave));
+        POOL.deposit(address(token), amountToAave, address(this), 0);
+        require(token.transferFrom(msg.sender,cruize_wallet, amountToCruize));
+    }
+
+    function depositToAave(address reserve, uint256 amount)
+        internal
+        returns (uint256)
+    {
+        if ((amount > 0 && msg.value != 0) || (msg.value > 0 && amount != 0))
```

Comment:
can you explain  what this condition  is doing ??

Reply:
we are checking ETH/ERC20 amounts, like if user is depositing ETH than msg.value will be greater than zero and amount for ERC20 will zero and vice versa.

Reply:
but is this. a good idea to do so like you are accepting an argument that represents amount means what amount 0f token will deposit. and you checking differently  for the erc20 token and ETH it's seems not an good idea